quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/Classes.java

```diff
@@ -50,13 +55,74 @@
     static final String RUNTIME_SUSPEND_EXECUTION_NAME = "co/paralleluniverse/fibers/RuntimeSuspendExecution";
     static final String UNDECLARED_THROWABLE_NAME      = UndeclaredThrowableException.class.getName().replace('.', '/');
     static final String SUSPEND_EXECUTION_NAME         = "co/paralleluniverse/fibers/SuspendExecution";
-    
-    // computed
-    static final String SUSPENDABLE_DESC = "Lco/paralleluniverse/fibers/Suspendable;";
-    static final String DONT_INSTRUMENT_DESC = Type.getDescriptor(DontInstrument.class);
-    static final String INSTRUMENTED_DESC = "Lco/paralleluniverse/fibers/Instrumented;";
+
     static final String LAMBDA_METHOD_PREFIX = "lambda$";
 
+    // CORE-21 : Provide getter and setter for annotation types.
+    static class TypeDesc {
+
+        enum ID {
+            SUSPENDABLE,
+            DONT_INSTRUMENT,
+            INSTRUMENTED,
+        };
+
+        // On TYPE_DESC_ID can map to more than one type.
+        private EnumMap<ID, Set<String>> descIds = new EnumMap<>(ID.class);
+
+        TypeDesc() {
+            set(ID.SUSPENDABLE, Type.getDescriptor(Suspendable.class));
+            set(ID.DONT_INSTRUMENT, Type.getDescriptor(DontInstrument.class));
+            set(ID.INSTRUMENTED, Type.getDescriptor(Instrumented.class));
```

Comment:
We were _explicitly_ avoiding using `Type.getDescriptor()` here, partly because it creates an unwanted dependency on the `fibers` package but also because it forces the JVM to load these classes.

The `DontInstrument` annotation lives in the `instrument` package and so hadn't caused any problems - yet.